src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -966,47 +966,76 @@ bool Literal::is_string_like() const noexcept {
 Literal Literal::add(Literal const &other, Node::NodeStorage &node_storage) const noexcept {
     return this->numeric_binop_impl([](auto const &num_ops) noexcept {
         return num_ops.add_fptr;
-    }, other, node_storage);
+    },
+                                    other, node_storage);
 }
 
 Literal Literal::operator+(Literal const &other) const noexcept {
     return this->add(other);
 }
 
+Literal &Literal::operator+=(const Literal &other) noexcept {
+    auto result = *this + other;
```

Comment:
We also need `Literal &Literal::{add,sub,...}_assign(Literal const &other, NodeStorage &node_storage)` (like `Literal::{add,sub,...}`) so one can directly specify the node storage where the new literal is supposed to end up.